todo/views.py

```diff
@@ -1,6 +1,73 @@
 # -*- coding: utf-8 -*-
 from __future__ import unicode_literals
 
-from django.shortcuts import render
+from django.contrib.auth.forms import UserCreationForm
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.http import HttpResponseRedirect
+from django.shortcuts import redirect
+from django.urls import reverse_lazy
+from django.views.generic import ListView, DetailView, FormView, DeleteView, CreateView
+
+from .forms import TaskForm
+from .models import Task
+
 
 # Create your views here.
+
+class AbstractLogin(LoginRequiredMixin):
+    login_url = '/login/'
+    redirect_field_name = reverse_lazy('tasklist')
+
+    class Meta:
+        abstract = True
+
+
+class TaskListView(AbstractLogin, ListView):
+    model = Task
+
+    def get_context_data(self, **kwargs):
+        context = super(TaskListView, self).get_context_data(**kwargs)
+        tasks = Task.objects.all()
+        filtered_tasks = [tasks.filter(status="todo"), tasks.filter(status="in_progress"), tasks.filter(status="done")]
+        context["tasks"] = filtered_tasks
+        return context
+
+
+class UpdateStatusView(AbstractLogin, DetailView):
+    model = Task
```

Comment:
I have to override get UpdateView from generic View have intermediate page. Can I do this better. Similiar problem in DeleteTaskView

Reply:
make normal form in template, it's POST action, don't mix them